guardian/production-metrics-lambdas
# Remove log4j

## What does this change?

Removes `log4j` in favour of `slf4j-simple`.

## How to test

Running `sbt dependencyTree | grep log4j` should not yield `log4j` as a dependency.

The logs we receive should be similar to the previous implementation – we do lose line numbers, a tradeoff for a simpler logging solution that avoids log4j (see discussion below).

Before:

```
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:59 - Content API response status: ok.
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:59 - Content API response status: ok.
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:67 - Total number of articles: 259
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:107 - Failed to transform CAPI article to CapiData with ID: crosswords/2022/jan/04/annotated-solutions-for-genius-222
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:107 - Failed to transform CAPI article to CapiData with ID: guardian-masterclasses/2022/jan/04/how-to-write-authentic-dialogue-a-practical-workshop-with-emma-claire-sweeney
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:107 - Failed to transform CAPI article to CapiData with ID: business/2022/jan/04/swedbank-ex-chief-birgitte-bonnesen-charged-baltics-money-laundering-scandal
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:107 - Failed to transform CAPI article to CapiData with ID: lifeandstyle/2022/jan/04/sudoku-5496-medium
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:107 - Failed to transform CAPI article to CapiData with ID: crosswords/cryptic/28645
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:107 - Failed to transform CAPI article to CapiData with ID: crosswords/quick/16118
2022-01-05 08:55:00 <> INFO  CapiAPILogic$:72 - Number of articles processed: 253
2022-01-05 08:55:07 <> INFO  CapiMetricsLambda:18 - Running the Capi Metrics Lambda.
```

After:

```
[INFO] metricsLambdas.logic.CapiAPILogic$ - Content API response status: ok.
[INFO] metricsLambdas.logic.CapiAPILogic$ - Content API response status: ok.
[INFO] metricsLambdas.logic.CapiAPILogic$ - Total number of articles: 259
[INFO] metricsLambdas.logic.CapiAPILogic$ - Failed to transform CAPI article to CapiData with ID: crosswords/2022/jan/04/annotated-solutions-for-genius-222
[INFO] metricsLambdas.logic.CapiAPILogic$ - Failed to transform CAPI article to CapiData with ID: guardian-masterclasses/2022/jan/04/how-to-write-authentic-dialogue-a-practical-workshop-with-emma-claire-sweeney
[INFO] metricsLambdas.logic.CapiAPILogic$ - Failed to transform CAPI article to CapiData with ID: business/2022/jan/04/swedbank-ex-chief-birgitte-bonnesen-charged-baltics-money-laundering-scandal
[INFO] metricsLambdas.logic.CapiAPILogic$ - Failed to transform CAPI article to CapiData with ID: lifeandstyle/2022/jan/04/sudoku-5496-medium
[INFO] metricsLambdas.logic.CapiAPILogic$ - Failed to transform CAPI article to CapiData with ID: crosswords/cryptic/28645
[INFO] metricsLambdas.logic.CapiAPILogic$ - Failed to transform CAPI article to CapiData with ID: crosswords/quick/16118
[INFO] metricsLambdas.logic.CapiAPILogic$ - Number of articles processed: 253
[INFO] metricsLambdas.CapiMetricsLambda - Running the Capi Metrics Lambda.
```

Note that Cloudwatch will still give us timestamps in an AWS context.

The lambda runs once a day, so testing via the normal route would mean waiting for the invocation ([logs](https://logs.gutools.co.uk/s/editorial-tools/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15d,to:now))&_a=(columns:!(app),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'67604290-baa6-11e9-bea2-633437abb232',key:app,negate:!f,params:(query:production-metrics-lambdas-capi-query),type:phrase),query:(match_phrase:(app:production-metrics-lambdas-capi-query)))),index:'67604290-baa6-11e9-bea2-633437abb232',interval:auto,query:(language:lucene,query:'stage:CODE'),sort:!())). Alternatively, we could invoke the lambda manually from AWS and watch the logs come in.
